client/html/themes/aimeos.js

```diff
@@ -1231,11 +1231,40 @@ AimeosCatalogList = {
 	},
 
 
+	/**
+	 * Enables infinite scroll if available
+	 */
+	setupInfiniteScroll: function() {
+		if( $('.catalog-list-items').data('infinite-url') != '') {
+			$(window).on( 'scroll', function() {
+				var infiniteUrl = $('.catalog-list-items').data('infinite-url');
+				
+				if( infiniteUrl != '' && $('.catalog-list-items')[0].getBoundingClientRect().bottom - $(window).height() < 50 ) {
+					$('.catalog-list-items').data('infinite-url', '');
```

Comment:
Wouldn't it be better to remove that line and overwrite `infinite-url` only when retrieving the next page succeeded (in case of a transport error)?

Reply:
Not sure about this. It might trigger a lot of XHR requests when a user scrolls the page multiple times before the new data is recieved - or some kind of Mutex would be necessary. I could re-write the original URL to the field in case of a failed Ajax request.

Reply:
OK, that's a good point I haven't thought at